src/promptflow/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -64,19 +44,32 @@ def trace_collector():
                     # TODO: persist with batch
                     span = Span._from_protobuf_object(span, resource=resource)
                     span._persist()
-                    if _is_workspace_configured():
-                        _write_trace_to_cosmosdb(span, line_summary_client, span_client)
+                    _try_write_trace_to_cosmosdb(span)
         return "Traces received", 200
 
     # JSON protobuf encoding
     elif "application/json" in content_type:
         raise NotImplementedError
 
 
-def _write_trace_to_cosmosdb(span: Span, line_summary_client, span_client):
+def _try_write_trace_to_cosmosdb(span: Span):
+    span_resource = span._content.get(SpanFieldName.RESOURCE, {})
+    resource_attributes = span_resource.get(SpanResourceFieldName.ATTRIBUTES, {})
+    subscription_id = resource_attributes.get(SpanResourceAttributesFieldName.SUBSCRIPTION_ID, None)
+    resource_group_name = resource_attributes.get(SpanResourceAttributesFieldName.RESOURCE_GROUP_NAME, None)
+    workspace_name = resource_attributes.get(SpanResourceAttributesFieldName.WORKSPACE_NAME, None)
+    if subscription_id is None or resource_group_name is None or workspace_name is None:
+        current_app.logger.info("Cannot find workspace info in span resource, skip writing to cosmosdb.")
```

Comment:
maybe debug is better? otherwise, there will be many info lines for this?

Reply:
Fixed